governator-core/src/main/java/com/netflix/governator/ModuleBuilder.java

```diff
@@ -0,0 +1,238 @@
+     * when the same module is installed in multiple placed. 
+     * @param modules
+     */
+    public ModuleBuilder overrideWith(Module ... modules) {
+        return overrideWith(Arrays.asList(modules));
+    }
+    
+    /**
+     * @see ModuleBuilder#overrideWith(Module...)
+     */
+    public ModuleBuilder overrideWith(Collection<Module> modules) {
+        this.module = Modules.override(module).with(modules);
+        return this;
+    }
+    
+    /**
+     * Add additional bindings to the module tracked by the DSL
+     * @param modules
+     */
+    public ModuleBuilder combineWith(Module ... modules) {
+        List<Module> m = new ArrayList<>();
+        m.add(module);
+        m.addAll(Arrays.asList(modules));
+        this.module = Modules.combine(m);
+        return this;
+    }
+    
+    /**
+     * Iterator through all elements and invoke a visitor.  This method
+     * is mean for side-effect free operations such as logging the current 
+     * state of bindings.
+     * @param visitor
+     */
+    public ModuleBuilder forEachElement(ElementVisitor<Void> visitor) {
+        for (Element element : Elements.getElements(module)) {
+            element.acceptVisitor(visitor);
+        }
+        return this;
+    }
+    
+    /**
+     * Log a warning that static injection is being used.  Static injection is considered a 'hack'
+     * to alllow for backwards compatibility with non DI'd static code.
+     */
+    public ModuleBuilder warnOfStaticInjections() {
+        return forEachElement(new WarnOfStaticInjectionVisitor());
+    }
+
+    /**
+     * Log the current binding state.  Log() is useful for debugging a sequence of
+     * operation where the binding snapshot can be dumped to the log after an operation.
+     * 
+     * @param prefix String prefix appended to each log message
+     */
+    public ModuleBuilder logEachKey(final String prefix) {
+        return forEachElement(new KeyTracingVisitor(prefix));
+    }
+    
+    /**
+     * @see {@link ModuleBuilder#logEachKey(String prefix)}
+     */
+    public ModuleBuilder logEachKey() {
+        return logEachKey("");
+    }
+    
+    /**
+     * Log a single 'info' line.  Call info in conjunction with calls to other logging calls
+     * to include headers and footers in the log.
+     * @param message
+     * @return
+     */
+    public ModuleBuilder info(String message) {
+        LOG.info(message);
+        return this;
+    }
+    
+    /**
+     * Extend the CORE dsl by providing a custom ModuleTransformer
+     * @param transformer
+     */
+    public ModuleBuilder transform(ModuleTransformer transformer) {
+        this.module = transformer.transform(module);
+        return this;
+    }
+    
+    /**
+     * Filter out elements for which the provided visitor returns true
+     * @param visitor
+     */
+    public ModuleBuilder filter(ElementVisitor<Boolean> visitor) {
```

Comment:
filter everywhere I've used it *keeps* the elements that match.  For example https://www.safaribooksonline.com/library/view/scala-cookbook/9781449340292/ch10s18.html